helix-rest/src/main/java/org/apache/helix/rest/client/CustomRestClientFactory.java

```diff
@@ -44,14 +51,17 @@ public static CustomRestClient get() {
         if (INSTANCE == null) {
           try {
             HttpClient httpClient;
+            RequestConfig config = RequestConfig.custom().setConnectTimeout(_httpRequestTimeout)
+                .setConnectionRequestTimeout(_httpRequestTimeout)
+                .setSocketTimeout(_httpRequestTimeout).build();
```

Comment:
Do we want to set the socket and connect timeout as well? Those two seem unrelated to the problem you're solving. I am not saying we shouldn't necessarily, but we should understand the implications of setting such configs.
Also it would be good to add a line of comment describing what you're doing here.

Reply:
I think we prefer to have a timeout for any reason and not be blocked to query SN. That is why I used both of them as same value. Also I noticed that we do same for other classes such as HttpRoutingDataReader. I do not have strong preferences about this. I feel we should reduce the amount of configs as much as possible.